java-sdk/sdk-checkpoints.asciidoc

```diff
@@ -0,0 +1,88 @@
+.. If the checkpoint data matches current data, the fetcher will emit the same checkpoint.
+. The SDK controller will process the candidates and update the checkpoint data in the SDK CrawlDB.
+
+
+=== Stopping a Running Job
+
+When a job is stopped, the current state of the job is stored so that it can be completed when the job is resumed.
+
+==== Stop Handling Design
+
+The SDK controller will keep track of the incomplete/complete items during a job. An incomplete item is an item that was emitted as a candidate but has not been sent to the fetcher to be processed. Alternatively, the fetch implementation may not have emitted the FetchResult message for that item. The incomplete item is stored in the SDK CrawlDB and marked as incomplete.
+
+A complete item is an item that was emitted first as a candidate and sent to the fetcher to be processed. The fetcher completes the process by sending a FetchResult message to indicate the process is finished. This item is then marked as complete by the SDK controllers by setting the `blockID` field in the item metadata to match the blockID of the current job.
+
+The blockId is used to quickly identify items in the CrawlDB which may not have been fully processed (completed). A blockId identifies a series of one or more Jobs. The lifetime of a blockId spans from the start of a crawl to the completion. The SDK controller will generate and use a new blockId when:
+
+* The current job is the first job.
+* The previous job’s state is `FINISHED`.
+
+When a job starts and the previous job did not complete, the previous job's state is `STOPPED`. In this case, the previous job's blockId is reused. The same blockId will be reused until the crawl successfully completes. The SDK controller will continue checking the CrawlDB for incomplete items, which are identified by having a blockID that doesn't match the previous job blockID. This approach ensures all items within the job are completed before the next job beings, even if the job was stopped multiple times before completion.
+
+When an item is considered a new candidate, the item's blockId does not change. Later, when the item is fully processed by the fetcher, the blockId is added to the item metadata and stored in the SDK CrawlDB. The item is then considered complete but will only be sent to fetchers when a new blockId is generated.
+
+When all items are complete, the SDK will check for checkpoints, as detailed in <<checkpoint-design, Checkpoint Design>>.
+
+IMPORTANT: A job won’t be started if there are incomplete items from the previous job stored in the SDK CrawlDB.
+
+===== Item Completion Flow
+image:/assets/images/sdkcheck-itemcomplete.png[Item Completion Flow]
+
+. The SDK controller gets a `FetchInput` from the SDK CrawlDB and sends it to the fetcher.
+. The fetcher receives the FetchInput.
+. The fetcher emits a candidate: `Item A`.
```

Comment:
Please change the arrow label from `Document` to `Candidate`. In that third step the items that is emitted is a Candidate.